### PR TITLE
support read write seperation

### PR DESCRIPTION
**Pull request by earayu**
## Description
Currently, Vitess allows users to specify the tablet type to access using the use command, such as use dbName@PRIMARY, use dbName@REPLICA. However, this requires users to modify their program to decide which type of tablet to use. If Vitess could make this decision automatically, it would make Vitess easier to use and better balance the load on different MySQL nodes.

## Related Issue(s)

[* RFC：https://github.com/vitessio/vitess/issues/12570](https://github.com/vitessio/vitess/issues/12570)

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by deepthi** [comment]
Hello! You seem to have put a significant amount of time into this. For future reference, we (maintainers) prefer to discuss the proposal first, especially when it is something radical like this, before a PR is created in order to avoid wasted time if we decide not to accept the proposal.

**Comment by earayu** [comment]
> Hello! You seem to have put a significant amount of time into this. For future reference, we (maintainers) prefer to discuss the proposal first, especially when it is something radical like this, before a PR is created in order to avoid wasted time if we decide not to accept the proposal.

Sure, you are right. 
I believe 'read write seperation' is a requirement that many needs, this PR could be a Proof of Concept. I am new to the Vitess commuity, so I am not sure what's the best way to propose a new feature. But I've attached a RFC for this feature: https://github.com/vitessio/vitess/issues/12570 
Maybe we can talk about the feature first

File: go/vt/sqlparser/analyzer_sql_type.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2021 The Vitess Authors.
```
Comment (harshit-gangal) [inline go/vt/sqlparser/analyzer_sql_type.go:2]: 2021 -> 2023

**Comment by harshit-gangal** [comment]
The RFC says comment directive is `NODE_TYPE` but the implementation has `ROLE`. I would like to suggest to use `TABLET_TYPE`.

Another interesting this is that at the table level it is already supported.
what I means is that if you do `select 1 from tbl@replica` this query will go to `REPLICA` tablet.

**Pull request closed without merging**